packages/block-library/src/image/edit.js

```diff
@@ -183,30 +183,41 @@ class ImageEdit extends Component {
 			isEditing: false,
 		} );
 
+		const mediaAttributes = pickRelevantMediaFiles( media );
+		const linkAttributes = this.buildLinkAttributes(
+			this.props.attributes.linkDestination,
+			{ ...this.props.attributes, ...mediaAttributes }
+		);
+
 		this.props.setAttributes( {
-			...pickRelevantMediaFiles( media ),
+			...mediaAttributes,
+			...linkAttributes,
 			width: undefined,
 			height: undefined,
 		} );
 	}
 
 	onSetLinkDestination( value ) {
+		this.props.setAttributes( this.buildLinkAttributes( value, this.props.attributes ) );
+	}
+
+	buildLinkAttributes( destination, attributes ) {
 		let href;
 
-		if ( value === LINK_DESTINATION_NONE ) {
+		if ( destination === LINK_DESTINATION_NONE ) {
 			href = undefined;
-		} else if ( value === LINK_DESTINATION_MEDIA ) {
-			href = ( this.props.image && this.props.image.source_url ) || this.props.attributes.url;
-		} else if ( value === LINK_DESTINATION_ATTACHMENT ) {
-			href = this.props.image && this.props.image.link;
+		} else if ( destination === LINK_DESTINATION_MEDIA ) {
+			href = attributes.source_url || attributes.url;
+		} else if ( destination === LINK_DESTINATION_ATTACHMENT ) {
+			href = attributes.link;
```

Comment:
The image block doesn't have `link` or `source_url` attributes. It looks like these should be `this.props.image.link` and `this.props.image.source_url`

Reply:
Hmm, I get an `this.props.image is null` error. I think this is again a case of the information not yet being available to set the link target like before, and it'll probably need another hack to get it right.

I'm starting to think this is a lot more work than it's worth. It's unfortunate that this default behaviour cannot be changed easily, because I believe having an image link to the larger version is a sane default that many users will want (indeed, it's the default in the classic editor, so it's what they'd also probably expect).

I might look into this again at some point but I can't dedicate much more time to fixing this right now. If someone else wants to have a go, please feel free!

Reply:
I agree with you. I've come to this pull request while I was searching for a way to change the default behaviour of the Gutenberg image block.